src/compiler/gcc.rs

```diff
@@ -420,9 +426,17 @@ where
         // We can't cache compilation without an input.
         None => cannot_cache!("no input file"),
     };
-    if language == None {
-        language = Language::from_file_name(Path::new(&input));
-    }
+    let language = match language {
+        None => {
+            let mut lang = Language::from_file_name(Path::new(&input));
+            // g++ and clang++ override the language to c++ if the -stdlib argument is passed.
```

Comment:
I think this should be "the -std argument" or something like that?

Actually, is that even right, because the behavior that the code is actually implementing is "if the compiler is g++ or clang++, parse it as C++ regardless of the file extension".  So perhaps we should just drop this comment entirely?